src/Sarif.Converters/ClangTidyConverter.cs

```diff
@@ -0,0 +1,171 @@
+            Location location = new Location()
+            {
+                PhysicalLocation = physicalLocation
+            };
+
+            result.Locations = new List<Location>()
+            {
+                location
+            };
+
+            if (entry.DiagnosticMessage.Replacements.Count > 0)
+            {
+                IList<Replacement> replacements = new List<Replacement>();
+
+                foreach (ClangTidyReplacement fix in entry.DiagnosticMessage.Replacements)
+                {
+                    Replacement replacement = new Replacement();
+
+                    replacement.DeletedRegion = new Region
+                    {
+                        CharLength = fix.Length,
+                        CharOffset = fix.Offset,
+                    };
+
+                    if (!string.IsNullOrEmpty(fix.ReplacementText))
+                    {
+                        replacement.InsertedContent = new ArtifactContent
+                        {
+                            Text = fix.ReplacementText
+                        };
+                    }
+
+                    replacements.Add(replacement);
+                }
+
+                var sarifFileChange = new ArtifactChange
+                {
+                    ArtifactLocation = new ArtifactLocation
+                    {
+                        Uri = analysisTargetUri
+                    },
+                    Replacements = replacements
+                };
+
+                Fix sarifFix = new Fix(description: null, artifactChanges: new List<ArtifactChange>() { sarifFileChange }, properties: null);
+                result.Fixes = new List<Fix> { sarifFix };
+            }
+
+            return result;
+        }
+
+        private static (List<ReportingDescriptor>, List<Result>) ExtractRulesAndResults(ClangTidyLog log)
+        {
+            var rules = new List<ReportingDescriptor>();
+            var ruleIds = new HashSet<string>();
+            var results = new List<Result>();
+
+            foreach (ClangTidyDiagnostic diagnostic in log.Diagnostics)
+            {
+                string ruleId = diagnostic.DiagnosticName;
+                (ReportingDescriptor, Result) ruleAndResult = SarifRuleAndResultFromClangTidyDiagnostic(diagnostic);
+                if (!ruleIds.Contains(ruleId))
+                {
+                    rules.Add(ruleAndResult.Item1);
+                    ruleIds.Add(ruleId);
+                }
+
+                results.Add(ruleAndResult.Item2);
+            }
+
+            return (rules, results);
+        }
+
+        private static (ReportingDescriptor, Result) SarifRuleAndResultFromClangTidyDiagnostic(ClangTidyDiagnostic diagnostic)
+        {
+            var reportingDescriptor = new ReportingDescriptor
+            {
+                Id = diagnostic.DiagnosticName,
+                DefaultConfiguration = new ReportingConfiguration
+                {
+                    Level = FailureLevel.Warning,
+                },
```

Comment:
I  think u dont need to add this.

Reply:
sorry did not get you, removing 
},
will have compile error

Reply:
remove entire DefaultConfiguration

Reply:
oh do you mean remove the whole DefaultConfiguration since no need. I am fixing it.

DefaultConfiguration = new ReportingConfiguration
                {
                    Level = FailureLevel.Warning,
                },

Reply:
fixed

Reply:
warning  is default. so we dont need to express.